### ISSUE TITLE
EDIT statement causes PC-BASIC to crash when a protected file is loaded and `--strict-protect` is on

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#22]
https://sourceforge.net/p/pcbasic/discussion/bugs/thread/c42d61f4/

To reproduce the bug, simply create any sort of program, even a simple
```
10 print "Hello world!"
```
will do. Then, save it as a protected file:
```
save "HELLO",P
```
Finally, start PC-BASIC with --strict-protect, load the file, and try to use an EDIT statement:
```
LOAD "HELLO"
EDIT 10
```
PC-BASIC will crash. The expected result is probably "Illegal function call", however.

**Comment by robhagemans** [comment]
Infinite loop, keeps printing `Illegal function call` then trying again.

**Comment by robhagemans** [comment]
Fixed by 7d06d38 (15.08) and 201b593 (16.12)
